lib/assignment/widgets/assignment_detail_screen.dart

```diff
@@ -89,15 +89,15 @@ class _AssignmentDetailScreenState extends State<AssignmentDetailScreen>
       return null;
     }
 
-    return FancyCachedBuilder<Course>(
-      controller: courseId.controller,
-      builder: (context, course, _) {
+    return EntityBuilder<Course>(
+      id: courseId,
+      builder: handleEdgeCases((context, course, _) {
```

Comment:
We already handle loading manually 
```suggestion
      builder: handleError((context, course, _) {
```